com.unity.ml-agents/Runtime/Inference/BarracudaModelExtensions.cs

```diff
@@ -0,0 +1,360 @@
+            }
+        }
+
+        /// <summary>
+        /// Check if the model has discrete action outputs.
+        /// </summary>
+        /// <param name="model">
+        /// The Barracuda engine model for loading static parameters.
+        /// </param>
+        /// <returns>True if the model has discrete action outputs.</returns>
+        public static bool HasDiscreteOutputs(this Model model)
+        {
+            if (model == null)
+                return false;
+            if (model.UseDeprecated())
+            {
+                return (int)model.GetTensorByName(TensorNames.IsContinuousControlDeprecated)[0] == 0;
+            }
+            else
+            {
+                return model.outputs.Contains(TensorNames.DiscreteActionOutput) &&
+                    (int)model.GetTensorByName(TensorNames.DiscreteActionOutputShape)[0] > 0;
+            }
+        }
+
+        /// <summary>
+        /// Discrete action output size of the model. This is equal to the sum of the branch sizes.
+        /// </summary>
+        /// <param name="model">
+        /// The Barracuda engine model for loading static parameters.
+        /// </param>
+        /// <returns>Size of discrete action output.</returns>
+        public static int DiscreteOutputSize(this Model model)
+        {
+            if (model == null)
+                return 0;
+            if (model.UseDeprecated())
+            {
+                return (int)model.GetTensorByName(TensorNames.IsContinuousControlDeprecated)[0] > 0 ?
+                    0 : (int)model.GetTensorByName(TensorNames.ActionOutputShapeDeprecated)[0];
+            }
+            else
+            {
+                var discreteOutputShape = model.GetTensorByName(TensorNames.DiscreteActionOutputShape);
+                return discreteOutputShape == null ? 0 : (int)discreteOutputShape[0];
+            }
+        }
+
+        /// <summary>
+        /// Discrete action output tensor name of the model.
+        /// </summary>
+        /// <param name="model">
+        /// The Barracuda engine model for loading static parameters.
+        /// </param>
+        /// <returns>Tensor name of discrete action output.</returns>
+        public static string DiscreteOutputName(this Model model)
+        {
+            if (model == null)
+                return null;
+            if (model.UseDeprecated())
+            {
+                return TensorNames.ActionOutputDeprecated;
+            }
+            else
+            {
+                return TensorNames.DiscreteActionOutput;
+            }
+        }
+
+        /// <summary>
+        /// Check if the model uses deprecated output fields and should be handled differently.
+        /// </summary>
+        /// <param name="model">
+        /// The Barracuda engine model for loading static parameters.
+        /// </param>
+        /// <returns>True if the model uses deprecated output fields.</returns>
+        public static bool UseDeprecated(this Model model)
```

Comment:
I wonder if this is future proof. There will be some more deprecations in the future. Maybe using a model version or calling this `NoHybridSupport`?

Reply:
Agreed, but don't like the negative term. I changed it to `HasHybridSupport()`.